File: peerprep-questions/model/question.go
package model

import "go.mongodb.org/mongo-driver/bson/primitive"

type Question struct {
	Id          primitive.ObjectID `json:"id,omitempty" bson:"_id,omitempty"` // mongo uses _id in bson for their object id, json id is for our requests
	Title       string             `json:"title"`
	Description string             `json:"description"`
	Category    string             `json:"category"` // could maybe have category & complexity as some enum
	Complexity  string             `json:"complexity"`
}

type UpdateQuestionRequest struct {
	Title       string `json:"title"`
	Description string `json:"description"`
	Category    string `json:"category"`
	Complexity  string `json:"complexity"`
}

File: backend/models/Question.js
const mongoose = require("mongoose");

const questionSchema = new mongoose.Schema({
    question_id: {
        type: Number,
        required: true,
        unique: true
    },
    title: {
        type: String,
        required: true,
        unique: true
    },
    description: {
        type: String,
        required: true,
    },
    category: {
        type: [String],
        required: true
    },
    complexity: {
        type: String,
        required: true
    },
    web_link: {
        type: String,
        required: true
    },
});

const Question = mongoose.model("question", questionSchema);
module.exports = Question;
